src/java/org/apache/cassandra/index/sai/disk/v2/V2VectorIndexSearcher.java

```diff
@@ -127,44 +131,66 @@ private PostingList searchPosting(QueryContext context, Expression exp, Abstract
         int topK = OverqueryUtils.topKFor(limit, graph.getCompressedVectors());
         float[] queryVector = exp.lower.value.vector;
 
-        BitsOrPostingList bitsOrPostingList = bitsOrPostingListForKeyRange(context, keyRange, queryVector, topK);
-        if (bitsOrPostingList.skipANN())
-            return bitsOrPostingList.postingList();
+        BitsOrRowIds bitsOrRowIds = bitsOrPostingListForKeyRange(keyRange, queryVector, topK, false);
+        if (bitsOrRowIds.skipANN())
+            return bitsOrRowIds.postingList();
 
-        var vectorPostings = graph.search(queryVector, topK, exp.getEuclideanSearchThreshold(), limit, bitsOrPostingList.getBits(), context);
-        bitsOrPostingList.updateStatistics(vectorPostings.getVisitedCount());
+        var vectorPostings = graph.search(queryVector, topK, exp.getEuclideanSearchThreshold(), limit, bitsOrRowIds.getBits(), context);
+        bitsOrRowIds.updateStatistics(vectorPostings.getVisitedCount());
         return vectorPostings;
     }
 
+    @Override
+    public OrderIterator orderBy(Expression exp, AbstractBounds<PartitionPosition> keyRange, QueryContext context, int limit) throws IOException
```

Comment:
I think we can improve how we structure the common code across search and orderBy paths to reduce the possibility for error both now and for future maintainers.  (For instance: it looks like bruteForceOrdering lost the optimization in findTopApproximatePostings of doing a first cut using the in-memory compressed vectors.)

Fundamentally both paths need to (1) get the vector ordinals [the search], (2) convert ordinals to rowids [RowIdIterator or something like it], and (3) convert those to a useful result either as a RangeIterator or OrderIterator.  So let's keep as much as possible for (1) and (2) the same and standardize on a single place to do (3).  

I believe the "right" way to structure this is to have the index handle (1) and (2) and then do converting of the raw row/score results to RI / OI in this class.